bash_scripting_102/bash_scripting_102.md

```diff
@@ -0,0 +1,477 @@
+
+The `mv` command will move files from their current location to a new location. This command needs to know **what** you want to move and **where** you want to move it to.
+
+To move `blue_jay.txt` to `blue_animals` run the command:
+
+```
+mv blue_jay.txt blue_animals
+```
+
+Now you can use `cd` and `ls` to see that `blue_jay.txt` is now in `blue_animals`.
+
+Notice that we used the **relative** locations of `blue_animals` and the macaw file. We could also have used their global location. Let's see how we would use **global** locations to move the `blue_morpho.txt` file, currently in `blue_animals`, to the same location as the other animals.
+
+```
+mv ~/Downloads/learning_bash-main/blue_animals/blue_morpho.txt ~/Downloads/learning_bash-main
+```
+
+<div class = "warning">
+If you moved `learning_bash-main` out of your Downloads folder, you will need to use the file path to it instead of `~/Downloads`.
+</div>
+
+You can also move multiple files at a time by entering them one after another, as long as the directory you want them to end up in is at the end.
+
+```
+mv blue_and_yellow_macaw.txt blue_morpho.txt blue_animals
+```
+
+<div class = "help">
+Did you get a message informing you that "No such file or directory" exists? Make sure you are in the right directory with `cd ~/Downloads/learning_bash-main`.
+</div>
+
+We can also use the `*` to move all files and folders matching a certain pattern to a new location. Let's make a new folder containing all files pertaining to blue whales:
+
+```
+mkdir blue_whale_folder
+mv blue_whale.* blue_whale_folder
+```
+This will move both `blue_whale.txt` and `blue_whale.dat` into the folder `blue_whale_folder`.
+
+Similarly we could move all of the files and folders that start with the word blue into `blue_animals`.
+
+```
+mv blue* blue_animals
+```
+You will get an output warning you that one of the things you asked for cannot be done:
+
+```
+mv: rename blue_animals to blue_animals/blue_animals: Invalid argument
+```
+
+**This is okay!** In fact it is bash being smart enough to know that you can't move a directory to become a sub-directory of itself! Use `cd` and `ls` to check that all of the other files and folders that started with the characters `blue` are now in the `blue_animals` directory.
+
+### Rename files with `mv`
+
+While there is a specific function for renaming files and folders, the easiest way to rename a file is to again use the `mv` function.
+
+Maybe we want to rename `koala.txt` to be `koala_bear.txt`. The `mv` function lets us do this by simply moving the file currently named `koala.txt` to a new file in the same location called `koala_bear.txt`:
+
+```
+mv koala.txt koala_bear.txt
+```
+
+<div class = "options">
+You can also use the global location names for your files or folders. In the example above we could have gotten the same result with
+
+```
+mv ~/Downloads/learning_bash-main/koala.txt ~/Downloads/learning_bash-main/koala_bear.txt
```

Comment:
This might be a more global issue, but I wonder if it's going to be obvious to leaners that they can sideways scroll in the code boxes?

Reply:
Hmm, I could put something about that in the lesson prep? I think it will come up in a lot of modules, but it would be good to put a note here!